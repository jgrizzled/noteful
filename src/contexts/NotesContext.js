import React from 'react';

export default React.createContext({
  notes: [],
  folders: [],
  addNote: () => null,
  deleteNote: () => null,
  modifyNote: () => null,
  addFolder: () => null,
  deleteFolder: () => null,
  modifyFolder: () => null
});
