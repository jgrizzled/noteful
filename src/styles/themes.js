// light and dark themes

const baseTheme = {
  fontSize: {
    small: '0.85rem',
    normal: '1rem',
    large: '1.25rem',
    xLarge: '1.5rem',
    xxLarge: '2rem'
  },
  font: {
    main: "'Roboto', sans-serif;"
  },
  fontWeight: {
    normal: '400',
    bold: '700'
  },
  breakpoint: {
    medium: '600px',
    large: '1000px'
  }
};

const lightTheme = {
  ...baseTheme,
  color: {
    primary: '#2bbd7e',
    background: '#FFFFFF',
    surface: '#f2f2f2',
    error: '#B00020',
    onPrimary: '#000000',
    onBackground: '#000000',
    onSurface: '#000000',
    link: '#1890ff'
  },
  boxShadow: {
    normal: '0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24)',
    normalHover: '0 14px 28px rgba(0,0,0,0.25), 0 10px 10px rgba(0,0,0,0.22)'
  }
};

const darkTheme = {
  ...baseTheme,
  color: {
    primary: '#2bbd7e',
    background: '#121212',
    surface: '#333333',
    error: '#CF6679',
    onPrimary: '#000000',
    onBackground: '#FFFFFF',
    onSurface: '#FFFFFF',
    link: '#99ceff'
  },
  boxShadow: {
    normal:
      '0 1px 3px rgba(255, 255, 255, 0.25), 0 1px 2px rgba(255, 255, 255, 0.25)',
    normalHover:
      '0px 5px 14px rgba(255, 255, 255, 0.5), 0px 3px 5px rgba(255, 255, 255, 0.22)'
  }
};

export default {
  light: lightTheme,
  dark: darkTheme
};
