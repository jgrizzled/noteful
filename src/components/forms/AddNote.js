import React, { useState, useContext } from 'react';
import { useParams } from 'react-router-dom';
import styled from 'styled-components';
import BigButton from 'components/common/BigButton';
import FolderSelector from 'components/forms/FolderSelector';
import NotesContext from 'contexts/NotesContext';

const Form = styled.form`
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 1rem;
`;

const NameInput = styled.input`
  width: 15rem;
  margin: 0.5rem auto;
  padding: 0.25rem;
`;

const ContentInput = styled.textarea`
  width: 100%;
  height: 80%;
  margin: 1rem auto;
  padding: 1rem;
  line-height: 1.5rem;
`;

export default props => {
  const params = useParams();
  const { folders } = useContext(NotesContext);
  let folderId = folders[0].id;
  if (params.folderId) {
    const matchFolderId = folders.find(f => f.id === params.folderId);
    if (matchFolderId) folderId = matchFolderId;
  }
  const [state, setState] = useState({
    name: '',
    folderId,
    content: ''
  });

  const onChange = e => {
    const { name, value } = e.target;
    setState(prevState => ({ ...prevState, [name]: value }));
  };
  const handleSubmit = e => {
    e.preventDefault();
    const note = { ...state };
    console.log(note);
  };

  return (
    <Form onSubmit={e => handleSubmit(e)}>
      <div>
        <label htmlFor='name'>Name:&nbsp;</label>
        <NameInput
          name='name'
          id='name'
          value={state.name}
          onChange={e => onChange(e)}
        />
      </div>
      <div>
        <label htmlFor='folderId'>Folder:&nbsp;</label>
        <FolderSelector
          name='folderId'
          id='folderId'
          folderId={state.folderId}
          onChange={onChange}
        />
      </div>
      <ContentInput
        value={state.content}
        name='content'
        placeholder='start typing...'
        onChange={e => onChange(e)}
      />
      <BigButton type='submit'>Add Note</BigButton>
    </Form>
  );
};
