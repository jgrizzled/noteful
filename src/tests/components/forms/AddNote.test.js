import React from 'react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import themes from 'styles/themes';
import AddNote from 'components/forms/AddNote';
import NotesContext from 'contexts/NotesContext';

const component = (
  <BrowserRouter>
    <NotesContext.Provider value={notesContextValue}>
      <ThemeProvider theme={themes.light}>
        <AddNote />
      </ThemeProvider>
    </NotesContext.Provider>
  </BrowserRouter>
);

// smoke test
it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(component, div);
  ReactDOM.unmountComponentAtNode(div);
});

// shallow snapshot test
it('renders as expected', () => {
  const tree = renderer.create(component).toJSON();
  expect(tree).toMatchSnapshot();
});
