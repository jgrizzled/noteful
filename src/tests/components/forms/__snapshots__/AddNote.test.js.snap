// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<form
  className="sc-AykKF fGgBWA"
  onSubmit={[Function]}
>
  <div>
    <label
      htmlFor="title"
    >
      Title: 
    </label>
    <input
      className="sc-AykKG hTwmRJ"
      id="title"
      name="title"
      onChange={[Function]}
      required={true}
      value=""
    />
  </div>
  <div>
    <label
      htmlFor="folder_id"
    >
      Folder: 
    </label>
    <select
      className="sc-AykKD cMrhNb"
      name="folder_id"
      onChange={[Function]}
      value={0}
    >
      <option
        value={0}
      >
        test folder
      </option>
    </select>
  </div>
  <textarea
    className="sc-AykKH giSvea"
    name="content"
    onChange={[Function]}
    placeholder="start typing..."
    required={true}
    value=""
  />
  <button
    className="sc-AykKC idWUwc"
    type="submit"
  >
    Submit
  </button>
</form>
`;
